core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/WhineyFunction.java

```diff
@@ -24,7 +24,7 @@
         return function.apply(arg);
       } catch (IOException e) {
         logger.error("Ignoring error", e);
-        return null;
+        throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)